attribute-masking-service/src/main/java/uk/gov/gchq/palisade/service/attributemask/service/KafkaHealthIndicator.java

```
@@ -0,0 +1,102 @@
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(KafkaHealthIndicator.class);
+
+    @Value("${akka.kafka.consumer.kafka-clients.group.id}")
+    private String groupId;
+
+    private final AdminClient adminClient;
+
+    /**
+     * Requires the AdminClient to interact with Kafka
+     * @param adminClient of the cluster
+     */
+    public KafkaHealthIndicator(final AdminClient adminClient) {
+        this.adminClient = adminClient;
+    }
+
+    @Override
+    public Health getHealth(final boolean includeDetails) {
+        if (performCheck()) {
+            return Health.up().withDetail("group", this.groupId).build();
+        } else {
+            return Health.down().withDetail("group", this.groupId).build();
+        }
+    }
+
+    /**
+     * Health endpoint
+     * @return the {@code Health} object
+     */
+    @Override
+    public Health health() {
+        if (performCheck()) {
+            return Health.up().build();
+        } else {
+            return Health.down().build();
+        }
+    }
+
+    private boolean performCheck() {
+        try {
+            Map<String, ConsumerGroupDescription> groupDescriptionMap = this.adminClient.describeConsumerGroups(Collections.singletonList(this.groupId))
+                            .all()
+                            .get(1, TimeUnit.SECONDS);
+
+            ConsumerGroupDescription consumerGroupDescription = groupDescriptionMap.get(this.groupId);
+
+            LOGGER.debug("Kafka consumer group ({}) state: {}", groupId, consumerGroupDescription.state());
+
+            if (consumerGroupDescription.state() == ConsumerGroupState.STABLE) {
+                return consumerGroupDescription.members().stream()
+                        .noneMatch(member -> (member.assignment() == null || member.assignment().topicPartitions().isEmpty()));
+            }
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
+            Thread.currentThread().interrupt();
```

either log or rethrow the exception

---

I am deliberately swallowing the exception because I want it to propagate a failed status, not stop processing